### PR TITLE
build script changed to Nuke

### PR DESCRIPTION
**Pull request by joaomatossilva**

### CODE REVIEW

File: build/Build.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    // Auto-injection fields:
+    // [GitVersion] readonly GitVersion GitVersion;
+    //  - Semantic versioning. Must have 'GitVersion.CommandLine' referenced.
+    // [GitRepository] readonly GitRepository GitRepository;
+    //  - Parses origin, branch name and head from git config.
+    // [Parameter] readonly string MyGetApiKey;
+    //  - Returns command-line arguments and environment variables.
+
+    public override AbsolutePath ArtifactsDirectory => RootDirectory / "artifacts";
```
Comment (matkoch) [inline build/Build.cs]: This should already be the implementation: https://github.com/nuke-build/nuke/blob/master/source/Nuke.Core/NukeBuild.cs#L187

File: build/Build.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            .Executes(() =>
+            {
+                DotNetRestore(s => DefaultDotNetRestore);
+            });
+
+    Target Compile => _ => _
+            .DependsOn(Clean, Restore)
+            .Executes(() =>
+            {
+                DotNetBuild(s => s
```
Comment (matkoch) [inline build/Build.cs]: You can also use the default settings here: https://github.com/nuke-build/nuke/blob/master/source/Nuke.Common/Tools/DotNet/DotNetTasks.cs#L39

In your case you would only call `SetVersionSuffix` then.

File: build/Build.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                    .SetVersionSuffix($"rev{Revision:D4}")
+                    .SetConfiguration(Configuration));
+            });
+
+    Target Pack => _ => _
+        .DependsOn(Compile, Test)
+        .Executes(() =>
+        {
+            DotNetPack(s => s
```
Comment (matkoch) [inline build/Build.cs]: Also a default setting there: https://github.com/nuke-build/nuke/blob/master/source/Nuke.Common/Tools/DotNet/DotNetTasks.cs#L48

Could remove `SetNoBuild` and `SetConfiguration`.

File: build/Build.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                .SetOutputDirectory(ArtifactsDirectory)
+                .SetProject(RootDirectory / "src" / "DateTimeExtensions")
+                .SetNoBuild(true)
```
Comment (matkoch) [inline build/Build.cs]: Can be written as `EnableNoBuild`.

File: build/Build.cs (hunk 3 above)
Comment (joaomatossilva) [reply build/Build.cs]: Actually, when I removed the `SetNoBuild` the parameter stopped to be included. I had to add the EnableNoBuil

Comment (joaomatossilva) [reply build/Build.cs]: nevermind.. found the issue.

File: build/Build.cs [hunk 5]
```diff
@@ -44,32 +46,28 @@ class Build : NukeBuild
...
-                .SetConfiguration(Configuration));
+                .SetVersionSuffix(RevisionString));
         });
 
     Target Test => _ => _
-        .DependsOn(Compile)
+        .DependsOn(Clean, Restore)
         .Executes(() =>
         {
             DotNetTest(s => s
```
Comment (matkoch) [inline build/Build.cs]: Isn't this actually compiling? This kind of dependency (Compile depends on Test), looks very uncommon 😃

Comment (matkoch) [reply build/Build.cs]: That's better :shipit:

**Pull request merged**